ovs/lib/helpers/decorators.py

```diff
@@ -215,12 +217,33 @@ def update_value(key, append, value_to_update=None):
                     if item['kwargs'] == kwargs_dict:
                         job_counter += 1
                         if job_counter == 2:  # 1st job with same params is being executed, 2nd is scheduled for execution ==> Discard current
-                            log_message('Execution of task {0} {1} discarded because of identical parameters'.format(task_name, params_info))
-                            return None
+                            if async_task is True:  # Not waiting for other jobs to finish since asynchronously
+                                log_message('Execution of task {0} {1} discarded because of identical parameters'.format(task_name, params_info))
+                                return None
+
+                            # Let's wait for 2nd job in queue to have finished
+                            if persistent_client.exists(persistent_key):
+                                counter = 0
+                                while counter < timeout:
+                                    log_message('Task {0} {1} is waiting for similar tasks to finish - ({2})'.format(task_name, params_info, counter + 1))
+                                    values = persistent_client.get(persistent_key)['values']
```

Comment:
Maybe use `get` with `must_exist` set to `False`, so you can prevent race conditions and act correctly when it might not exist.